src/main/java/io/cdap/plugin/spark/SQLUtils.java

```diff
@@ -66,12 +79,15 @@ public static String getAdditionalFieldsForRoot(HierarchyToRelationalConfig conf
    * @return sql statement for fetching unique records
    */
   public static String sqlForUniqueRecords(HierarchyToRelationalConfig config, Row root,
-                                           String tableName) {
+                                           String tableName, boolean includeRootRecord) {
```

Comment:
update javadoc. 
`includeRootRecord ` should be `isSelfReferencingRoot`. That way the if condition can be reversed.